regimes/fi/tax_identity.go

```diff
@@ -0,0 +1,177 @@
+package fi
+
+import (
+	"errors"
+	"regexp"
+	"strconv"
+
+	"github.com/invopop/gobl/cbc"
+	"github.com/invopop/gobl/l10n"
+	"github.com/invopop/gobl/tax"
+	"github.com/invopop/validation"
+)
+
+// TaxCodeType represents the types of tax codes issued in Finland.
+type TaxCodeType string
+
+// Supported tax code types.
+const (
+	NationalTaxCode     TaxCodeType = "H" // Finnish Personal Identity Code (PIC) - (Henkilötunnus)
+	ForeignTaxCode      TaxCodeType = "V" // Veronumero; temporary workers, especially in construction or shipbuilding
+	OrganizationTaxCode TaxCodeType = "Y" // Y-tunnus
+	UnknownTaxCode      TaxCodeType = "NA"
+)
```

Comment:
Rather than defining a new type, I'd probably go for a `cbc.Code` to compare against.